evalml/pipelines/utils.py

```diff
@@ -99,6 +100,12 @@ def make_pipeline(X, y, estimator, problem_type, custom_hyperparameters=None, te
         class: PipelineBase subclass with dynamically generated preprocessing components and specified estimator
 
     """
+    X = _convert_to_woodwork_structure(X)
+    y = _convert_to_woodwork_structure(y)
+
+    X = _convert_woodwork_types_wrapper(X.to_pandas())
+    y = _convert_woodwork_types_wrapper(y.to_pandas())
```

Comment:
I don't think this actually needs to be converted - nothing in this function seems to explicitly need numpy types. `_get_preprocessing_components` uses dtype selection, can we use woodwork for that instead? This would also simplify the text support code, if `_get_preprocessing_components` can check for text columns itself rather than the variable being passed all the way through the stack trace from `search` (as per the discussion [here](https://github.com/alteryx/evalml/pull/1062#discussion_r512944810))

Alternatively, we could file a separate issue for this

Reply:
Yea I think it would be better for `_get_preprocessing_components` to have access to the woodwork table rather than a dataframe. I think we'll need to do this eventually because our logic to determine the pre-processing components should leverage the full range of logical types in ww. Might be best to do that sooner rather than later?

Also, as @eccabay mentioned, that would move us closer to being able to delete the `text_columns` parameter which I think is nice!

Reply:
Ah, good idea! I was planning on doing the connectors and passing woodwork through each of the methods in a separate PR but you guys are right that it's work that needs to be done at some point so better sooner than later. Updating, thanks all!